src/libraries/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.OpenSslAvailable.cs

```diff
@@ -0,0 +1,19 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Runtime.InteropServices;
+
+internal static partial class Interop
+{
+    internal static class OpenSslNoInit
+    {
+        [DllImport(Libraries.CryptoNative, EntryPoint = "CryptoNative_OpenSslAvailable")]
```

Comment:
I think instead of a new entrypoint, we could reuse `CryptoNative_EnsureOpenSslInitialized` as it returns the boolean and also takes care of non-portable builds.

Reply:
That was the thought I had as well, but we do want the Linux version to stay catastrophic, so this is probably a nicer approach.

Reply:
The issue with `EnsureOpenSslInitialized` is it calls `InitializeOpenSSLShim`, which will sigabrt if OpenSSL is not available.

Reply:
In a cctor we have: https://github.com/dotnet/runtime/blob/01b7e73cd378145264a7cb7a09365b41ed42b240/src/libraries/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.Initialization.cs#L34 so it will sigabrt on those platforms nonetheless, no?

aside, we can delete this include with the current change https://github.com/dotnet/runtime/blob/6b06471d2737ec2289deaa580a61f13566030cb0/src/libraries/Native/Unix/System.Security.Cryptography.Native/opensslshim.c#L8

Reply:
The property I added is in a `OpenSslNoInit` class that does not have a cctor which does the initialization.

Reply:
Yup, when `class OpenSsl`'s cctor is called it ends up calling `EnsureOpenSslInitialized`. When openssl is not available, it's a no-go on linux before and after this change for real-world applications. I can see that in unit tests we want to avoid such catastrophic sigabrt.